pkg/api/util/util.go

```diff
@@ -60,3 +83,35 @@ func Validate(w http.ResponseWriter, r *http.Request) error {
 
 	return err
 }
+
+// ValidateDCARequest is used for the exposed endpoints of the DCA.
+// It is different from Validate as we want to have different validations.
+func ValidateDCARequest(w http.ResponseWriter, r *http.Request) error {
+	var err error
+	auth := r.Header.Get("Authorization")
+	if auth == "" {
+		w.Header().Set("WWW-Authenticate", `Bearer realm="Datadog Agent"`)
+		err = fmt.Errorf("no session token provided")
+		http.Error(w, err.Error(), 401)
+		return err
+	}
+
+	tok := strings.Split(auth, " ")
+	if tok[0] != "Bearer" {
+		w.Header().Set("WWW-Authenticate", `Bearer realm="Datadog Agent"`)
+		err = fmt.Errorf("unsupported authorization scheme: %s", tok[0])
+		http.Error(w, err.Error(), 401)
+		return err
+	}
+	dcaToken := config.Datadog.GetString("cluster_agent.auth_token")
+	if dcaToken == "" {
+		dcaToken = GetDCAAuthToken()
+	}
+
+	if len(tok) < 2 || tok[1] != dcaToken {
```

Comment:
Q: Is it intend to accept the auth string if like:
`Bearer 32longstring anotherstring` 
Because the length of tok will be 3.

Reply:
Good catch, I used the mechanism from the agent - I'll update this.